inference/core/models/roboflow.py

```diff
@@ -583,89 +454,46 @@ def download_weights(self) -> None:
-                            )
-                            sleep(SLEEP_SECONDS_BETWEEN_RETRIES)
-                    if not success:
-                        raise Exception(f"Failed to download model artifacts.")
-            else:
-                # AWS Keys are not available so we use the API Key to hit the Roboflow API which returns a signed link for downloading model artifacts
-                self.api_url = ApiUrl(
-                    f"{API_BASE_URL}/core_model/{self.endpoint}?api_key={self.api_key}&device={self.device_id}&nocache=true"
-                )
-                api_data = get_api_data(self.api_url)
-                if "weights" not in api_data.keys():
-                    raise TensorrtRoboflowAPIError(
-                        f"An error occurred when calling the Roboflow API to acquire the model artifacts. The endpoint to debug is {self.api_url}. The error was: Key 'tensorrt' not in api_data."
-                    )
-
-                weights_url_keys = api_data["weights"].keys()
-
-                for weights_url_key in weights_url_keys:
-                    weights_url = ApiUrl(api_data["weights"][weights_url_key])
-                    t1 = perf_counter()
-                    attempts = 0
-                    success = False
-                    while attempts < 3 and not success:
-                        r = requests.get(weights_url)
-                        filename = weights_url.split("?")[0].split("/")[-1]
-                        file_path = self.open_cache(f"{filename}", "wb")
-                        with file_path as f:
-                            f.write(r.content)
-
-                        # Check file size
-                        if os.path.getsize(file_path.name) <= 4096:
-                            if attempts < 2:
-                                attempts += 1
-                                continue
-                            else:
-                                raise Exception(
-                                    f"Failed to download model artifacts from API after 2 attempts."
-                                )
-                        else:
-                            success = True
-
-                    if perf_counter() - t1 > 120:
-                        self.log(
-                            "Weights download took longer than 120 seconds, refreshing API request"
-                        )
-                        api_data = get_api_data(self.api_url)
-        else:
+        infer_bucket_files = self.get_infer_bucket_file_list()
+        if are_all_files_cached(files=infer_bucket_files, model_id=self.endpoint):
             self.log("Model artifacts already downloaded, loading from cache")
+            return None
+        if is_model_artefacts_bucket_available():
+            self.download_model_artefacts_from_s3()
+            return None
+        self.download_mode_from_roboflow_api()
+
+    def download_mode_from_roboflow_api(self) -> None:
```

Comment:
I think this is a typo? Should this be `download_model_from_roboflow_api`?

Reply:
👍